packages/x-date-pickers/src/internals/models/props/basePickerProps.tsx

```diff
@@ -52,6 +52,12 @@ export interface BaseNonStaticPickerProps {
    * Defaults to localized format based on the used `views`.
    */
   format?: string;
+  /**
+   * Density of the format when rendered in the input.
+   * When "normal", a space will be added before and after every `/` character to make the format more readable.
```

Comment:
What do you think about adding more locale/format agnostic description referring to a separator between different format tokens (e.g. `/` in US format) or something along these lines? 🤔

Reply:
If you have a suggestion I'm interested
I agree that being locale/format agnostic is better, but I struggle to write something easy to understand.

Reply:
What's the link with locale? For now, it only applies to `/`, and it would feel strange to apply it to other separators, such as `.` or `-`

```js
if (formatDensity === 'normal' && cleanedSeparator === '/') {
  cleanedSeparator = ' / ';
}
```

Reply:
Well damn, sorry for not exploring this conundrum deep enough. 🤷 
In that case, if we are looking from the user's perspective, you could say that the current case is simply a bug that needs to be solved.
Because this `formatDensity` prop seems somewhat like a biased monkey patch for formats that include a `/` token separator, whereas other locales work as expected... 🙈 
- format="YYYY/MM/DD" results in `YYYY / MM / DD` format
- format="YYYY-MM-DD" results in `YYYY-MM-DD` format
- format="YYYY.MM.DD" results in `YYYY.MM.DD` format
So, just saying, that I can concur, that from a user's perspective, it can be seen as simply a bug. 🤔

Reply:
I would personally be in favor of adding the space to any separator that is between two date sections or one date and one time section. 

____

We added the space in the first place because the designs of Gerda had them and because I do agree with here that it's more readable.

My assumptions for enabling it by default were:

1. The vast majority of people are either using the default formats (based on the view), 
2. If the spaces don't make the value overflow your input (i.e: if your application is not very dense), then having some space between the sections make the value more readable.

If you look at Spectrum rendering, there is some space between the sections (added by CSS, not with spaces since it's a multi HTML approach):

![image](https://user-images.githubusercontent.com/3309670/229082261-b9e4887d-08b7-4d79-8508-a06758336962.png)

Same for the native browser input (in Chrome at least):

![image](https://user-images.githubusercontent.com/3309670/229083025-6ac1649d-1345-4818-bb85-70a4e707a4a7.png)

Of course with our current single HTML input approach, we are limited in the size of the spaces we can apply.
The space between two sections on the native browser input seem to be smaller than ours (not sure though).
This topic will be easier to handle with the multi input approach.

Reply:
> we end with 4 digits years but in asia, dates format are YYYY-MM-DD which makes it more blurry

That's very true :/

Reply:
> I would personally be in favor of adding the space to any separator that is between two date sections or one date and one time section.

I don't disagree, but IMHO, it would be ideal if the spacing would only be visual, avoiding the need for end-user to think about re-formatting data/value.



> 1. The vast majority of people are either using the default formats (based on the view),

And even in that case, it only applies if your locale uses `/` as a token separator. Many countries do not. 🤔 



> Same for the native browser input (in Chrome at least):

My guess is that the native date input is just using some font trickery magic, as you can see that changing the `font-family` achieves nearly the same spacing result. 🤔 
![Screenshot 2023-03-31 at 12 50 19](https://user-images.githubusercontent.com/4941090/229099544-3bed5e92-b3a7-4cc2-a3b8-764f5c9f49c8.png)

Reply:
If we have a solution that do not change the format and only rely on CSS, that's way better I agree

Reply:
I'm not sure there is one with the current approach.
Anyways, I just wanted to bring a different perspective in, especially if you haven't looked at the issue from this point of view.
TL.DR. From my PoV (user position) the issue with token discrepancy is the biggest problem.
Going with either:
- remove spacing from all formats (aligning with locales using any separator other than `/`);
- add spacing to all formats (aligning with the most popular format using `/` separator);

In both cases add the `formatDensity` prop to allow for control over adding the optional spaces between the separator.

Reply:
> add spacing to all formats 

Not sure to understand what you mean by "all format" here

Reply:
> > add spacing to all formats
> 
> Not sure to understand what you mean by "all format" here

https://github.com/mui/mui-x/pull/8425#discussion_r1154282207

Reply:
OK, so adding the space also to `.` and `-` ?

I guess we have to differentiate two steps:

1. Make the non-breaking changes we can today to improve the consistency (adding the space to `.` and `-` can be one of them)
2. Discuss how we can make the overall structure even more consistent with breaking changes for v7 (but we need to decide if we go for multi HTML input before imho)

Reply:
I came up with a temporary solution of using [invisible, no space characters](https://invisibletext.com/) for the `format` string to bypass the `cleanedSeparator === '/'` check. Terrible, I know. 😅 

`format="MM‎/dd‎/yyy"`

![image](https://user-images.githubusercontent.com/220755/229937156-cc627100-959f-4acb-9964-9636d0abbd6b.png)